sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DistinctHash.java

```diff
@@ -3,60 +3,129 @@
-import com.azure.cosmos.models.ModelBridgeInternal;
-import com.fasterxml.jackson.databind.MapperFeature;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.ValueNode;
 
 import java.io.IOException;
-import java.nio.ByteBuffer;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Map;
 
 public final class DistinctHash {
 
-    private static final UInt128 ARRAY_HASH_SEED = new UInt128(0xfa573b014c4dc18eL, 0xa014512c858eb115L);
+    private static final UInt128 ROOT_HASH_SEED = new UInt128(0xbfc2359eafc0e2b7L, 0x8846e00284c4cf1fL);
 
-    private static final ObjectMapper OBJECT_MAPPER =
-        new ObjectMapper()
-            .configure(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY, true)
-            .configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true);
+    private static class HashSeeds {
+        public static final UInt128 NULL = new UInt128(0x1380f68bb3b0cfe4L, 0x156c918bf564ee48L);
+        public static final UInt128 FALSE = new UInt128(0xc1be517fe893b40cL, 0xe9fc8a4c531cd0ddL);
+        public static final UInt128 TRUE = new UInt128(0xf86d4abf9a412e74L, 0x788488365c8a985dL);
+        public static final UInt128 STRING = new UInt128(0x61f53f0a44204cfbL, 0x09481be8ef4b56ddL);
+        public static final UInt128 ARRAY = new UInt128(0xfa573b014c4dc18eL, 0xa014512c858eb115L);
+        public static final UInt128 OBJECT = new UInt128(0x77b285ac511aef30L, 0x3dcf187245822449L);
+        public static final UInt128 INTEGER = new UInt128(0x0320dc908e0d3e71L, 0xf575de218f09ffa5L);
+        public static final UInt128 LONG = new UInt128(0xed93baf7fdc76638L, 0x0d5733c37e079869L);
+        public static final UInt128 DOUBLE = new UInt128(0x62fb48cc659963a0L, 0xe9e690779309c403L);
+        public static final UInt128 ARRAY_INDEX = new UInt128(0xfe057204216db999L, 0x5b1cc3178bd9c593L);
+        public static final UInt128 PROPERTY_NAME = new UInt128(0xc915dde058492a8aL, 0x7c8be2eba72e4634L);
+    }
 
     @SuppressWarnings("unchecked")
     public static UInt128 getHash(Object resource) throws IOException {
+        return getHash(resource, ROOT_HASH_SEED);
+    }
 
-        if (resource instanceof List) {
-            return getHashFromList((List<Object>) resource);
+    @SuppressWarnings("unchecked")
+    private static UInt128 getHash(Object resource, UInt128 seed) throws IOException {
+        if (resource == null) {
+            return MurmurHash3_128.hash128(HashSeeds.NULL, seed);
         }
-
         if (resource instanceof JsonSerializable) {
-            return getHashFromJsonSerializable((JsonSerializable) resource);
+            return getHashFromJsonSerializable((JsonSerializable) resource, seed);
```

Comment:
I think there is a possible `infinite recursion` scenario here. 

This calls `getHashFromJsonSerializable` -> which then again calls this method `getHash` by casing resource to `JsonSerializable`

Reply:
it will return ObjectNode for the propertyBag, so should not have endless loop